### PR TITLE
SUM-253 | add heading size option

### PR DESCRIPTION
**Pull request by rebeccahongsf**
# READY FOR REVIEW
# Summary
- SUM-253 | add heading size option

# Review By (Date)
- When possible

# Review Tasks

## Setup tasks and/or behavior to test

1.  Review code

# Associated Issues and/or People
- SUM-253
- PR #67

### CODE REVIEW

File: src/components/paragraphs/stanford-card/card-paragraph.tsx [hunk 1]
```diff
@@ -14,6 +14,23 @@ type Props = HtmlHTMLAttributes<HTMLDivElement> & {
...
+  }
+  if (!headingTypes[heading] || !size || size === "default") return ""
+  let type = headingTypes[heading]
+  if (size === "larger") {
+    type += 1
+  } else if (size === "smaller") {
+    type -= 1
+  }
+
+  return `type-${type}`
```
Comment (pookmish) [inline src/components/paragraphs/stanford-card/card-paragraph.tsx:31]: remember, dynamic classes aren't reliable. use `clsx` here like 
```
clsx({
  "type-3": type === 3,
  "type-2": type === 2,
})
```

Optionally, we could even rewrite this entire logic to be just a bigger clsx call:

```
const isH2 = heading === "h2"
const isH3 = heading === "h3"
const isH4 = heading === "h4"
const isLarger = size === "larger"
const isSmaller = size === "smaller"

clsx({
  "type-3": (isH2 && !size) || (isH3 && isLarger) ... etc
  "type-2": (isH3 && !size) || (isH2 && isSmaller) ... etc
  "type-1": (isH4 && !size) .... etc
})
```

File: src/components/paragraphs/stanford-card/card-paragraph.tsx [hunk 2]
```diff
@@ -22,8 +39,12 @@ const CardParagraph = ({paragraph, ...props}: Props) => {
     paragraph.suCardMedia?.__typename === "MediaVideo" ? paragraph.suCardMedia.mediaOembedVideo : undefined
 
   const headerTagChoice = (behaviors.su_card_styles?.heading || "h2").split(".", 2)
+  const headerSize = behaviors.su_card_styles?.sum_card_heading_size
   const headerTag = headerTagChoice[0]
-  const headerClasses = headerTagChoice[1]?.replace(".", " ").replace("su-font-splash", "type-3 mb-12") || "mb-12"
+  const headerClasses = twMerge(
```
Comment (pookmish) [inline src/components/paragraphs/stanford-card/card-paragraph.tsx:44]: this is greatly increasing complexity, I think we advise against this in the future.

File: src/components/paragraphs/stanford-card/card-paragraph.tsx [hunk 3]
```diff
@@ -14,6 +14,23 @@ type Props = HtmlHTMLAttributes<HTMLDivElement> & {
...
+    h3: 3,
+    h4: 2,
+  }
+  if (!headingTypes[heading] || !size || size === "default") return ""
+  let type = headingTypes[heading]
+  if (size === "larger") {
+    type += 1
+  } else if (size === "smaller") {
+    type -= 1
+  }
```
Comment (pookmish) [inline src/components/paragraphs/stanford-card/card-paragraph.tsx:29]: at this point, we know that the `size` is either `larger` or `smaller`, so we can simplify the condition.

```suggestion
  const type = size === "larger" ? headingTypes[heading] + 1 : headingTypes[heading] - 1
```

**Review by pookmish — CHANGES_REQUESTED** [review]

**Comment by pookmish** [comment]
@rebeccahongsf see [this comment](https://github.com/SU-SWS/summer-nextjs/pull/66#discussion_r1765333947) first since it might push you one way or another here.

File: src/components/paragraphs/stanford-card/card-paragraph.tsx [hunk 4]
```diff
@@ -14,6 +14,18 @@ type Props = HtmlHTMLAttributes<HTMLDivElement> & {
   paragraph: ParagraphStanfordCard
 }
 
+const adjustHeadingType = (heading: "h2" | "h3" | "h4" | "div", size?: "larger" | "smaller"): string | number => {
+  const headingTypes: Record<string, number> = {
+    h2: 4,
+    h3: 3,
+    h4: 2,
+  }
+  if (!headingTypes[heading] || !size) return ""
```
Comment (pookmish) [inline src/components/paragraphs/stanford-card/card-paragraph.tsx:23]:
```suggestion
  if (!headingTypes[heading] || !size) return
```

File: src/components/paragraphs/stanford-card/card-paragraph.tsx [hunk 5]
```diff
@@ -14,6 +14,18 @@ type Props = HtmlHTMLAttributes<HTMLDivElement> & {
...
+  const type = size === "larger" ? headingTypes[heading] + 1 : headingTypes[heading] - 1
+
+  return type
```
Comment (pookmish) [inline src/components/paragraphs/stanford-card/card-paragraph.tsx:26]:
```suggestion
  return size === "larger" ? headingTypes[heading] + 1 : headingTypes[heading] - 1
```

File: src/components/paragraphs/stanford-card/card-paragraph.tsx [hunk 6]
```diff
@@ -14,6 +14,18 @@ type Props = HtmlHTMLAttributes<HTMLDivElement> & {
   paragraph: ParagraphStanfordCard
 }
 
+const adjustHeadingType = (heading: "h2" | "h3" | "h4" | "div", size?: "larger" | "smaller"): string | number => {
```
Comment (pookmish) [inline src/components/paragraphs/stanford-card/card-paragraph.tsx:17]:
```suggestion
const adjustHeadingType = (heading: "h2" | "h3" | "h4" | "div", size?: "larger" | "smaller"): number | undefined => {
```

**Review by pookmish — CHANGES_REQUESTED** [review]
just a small change, but I think this is better already

File: src/components/patterns/pill-card.tsx [hunk 7]
```diff
@@ -66,9 +66,11 @@ const PillCard = ({imageUrl, imageAlt, videoUrl, isArticle, children, bgColor, .
 
       <div
         className={twMerge(
-          "rs-px-3 flex flex-col pb-[125px] pt-20 @7xl:rs-px-4 @2xl:pb-[175px] @3xl:pb-[225px] @4xl:pb-[300px]",
+          "rs-px-3 flex flex-col pb-[125px] pt-20 @7xl:rs-px-4 @2xl:pb-[175px] @3xl:pb-[225px] @4xl:pb-[300px] [&_p>a]:text-black-true hocus:[&_p>a]:text-[#00365C]",
```
Comment (rebeccahongsf) [inline src/components/patterns/pill-card.tsx:69]: Slipping in the accessibility color change as well 🤔 Does this darker digital blue color change work for the hover/focus state?

https://webaim.org/resources/contrastchecker/?fcolor=00365C&bcolor=F9A44A
https://webaim.org/resources/contrastchecker/?fcolor=00365C&bcolor=A6B168
https://webaim.org/resources/contrastchecker/?fcolor=00365C&bcolor=F4795B

Comment (pookmish) [reply src/components/patterns/pill-card.tsx:69]: Looks like it does, but the links on the "Spirited Light" (2.46) and the "Olive Light" (2.86) still don't work

**Review by pookmish — CHANGES_REQUESTED** [review]
Gotta fix the link colors in two card variants.

https://github.com/SU-SWS/summer-nextjs/pull/66/files#r1765795628

**Comment by pookmish** [comment]
I hate to be the bearer of bad news, but we still have contrast issues with the hover color. "Olive Light" & "Spirited Light hav an opacity, so we gotta take that into consideration.
image

image

**Comment by rebeccahongsf** [comment]
I went with an even darker blue 👀 Does that suffice or shall I go back to the drawing board 🤔 

https://webaim.org/resources/contrastchecker/?fcolor=001829&bcolor=F9A44A
https://webaim.org/resources/contrastchecker/?fcolor=001829&bcolor=A6B168
https://webaim.org/resources/contrastchecker/?fcolor=001829&bcolor=F4795B
https://webaim.org/resources/contrastchecker/?fcolor=F4795BCC&bcolor=001829
https://webaim.org/resources/contrastchecker/?fcolor=A6B168CC&bcolor=001829

**Review by pookmish — APPROVED** [review]

**Pull request merged**